src/Temporalio/Worker/Tuning/WorkerTuner.cs

```diff
@@ -0,0 +1,106 @@
+        /// </summary>
+        /// <param name="targetMemoryUsage">A value between 0 and 1 that represents the target
+        /// (system) memory usage. It's not recommended to set this higher than 0.8, since how much
+        /// memory a workflow may use is not predictable, and you don't want to encounter OOM
+        /// errors.</param>
+        /// <param name="targetCpuUsage">A value between 0 and 1 that represents the target (system)
+        /// CPU usage. This can be set to 1.0 if desired, but it's recommended to leave some
+        /// headroom for other processes.</param>
+        /// <param name="workflowOptions">Options for the workflow task slot supplier.</param>
+        /// <param name="activityOptions">Options for the activity task slot supplier.</param>
+        /// <param name="localActivityOptions">Options for the local activity slot supplier.</param>
+        /// <returns>The tuner.</returns>
+        /// <remarks>
+        /// WARNING: Resource based tuning is currently experimental.
+        /// </remarks>
+        public static WorkerTuner CreateResourceBased(
+            double targetMemoryUsage,
+            double targetCpuUsage,
+            ResourceBasedSlotSupplierOptions? workflowOptions = null,
+            ResourceBasedSlotSupplierOptions? activityOptions = null,
+            ResourceBasedSlotSupplierOptions? localActivityOptions = null)
+        {
+            ResourceBasedTunerOptions tunerOpts =
+                new ResourceBasedTunerOptions(targetMemoryUsage, targetCpuUsage);
+            return new(
+                new ResourceBasedSlotSupplier(
+                    workflowOptions == null
+                        ? new ResourceBasedSlotSupplierOptions()
+                        : workflowOptions,
+                    tunerOpts),
+                new ResourceBasedSlotSupplier(
+                    activityOptions == null
+                        ? new ResourceBasedSlotSupplierOptions()
+                        : activityOptions,
+                    tunerOpts),
+                new ResourceBasedSlotSupplier(
+                    localActivityOptions == null
+                        ? new ResourceBasedSlotSupplierOptions()
+                        : localActivityOptions,
+                    tunerOpts));
+        }
+
+        /// <summary>
+        /// Create a fixed-size tuner with the given slot capacities.
+        /// </summary>
+        /// <param name="numWorkflowTaskSlots">The number of available workflow task slots.</param>
+        /// <param name="numActivityTaskSlots">The number of available activity task slots.</param>
+        /// <param name="numLocalActivitySlots">The number of available local activity slots.</param>
+        /// <returns>The tuner.</returns>
+        public static WorkerTuner CreateFixedSize(
+            int numWorkflowTaskSlots,
+            int numActivityTaskSlots,
+            int numLocalActivitySlots)
```

Comment:
```suggestion
            int workflowTaskSlots,
            int activityTaskSlots,
            int localActivitySlots)
```

Not a big deal, but if you do end up making another commit, might as well (or suffix with `Count` or whatever). Matters a bit more here than in Java because .NET supports named parameters.